processors/enduserprocessor/enduserprocessor.go

```diff
@@ -0,0 +1,535 @@
+	}
+
+	if user == nil {
+		return
+	}
+
+	if len(user.id) > 0 {
+		addSpanAttribute(span, enduserIDAttribute, user.id)
+	}
+
+	if len(user.role) > 0 {
+		addSpanAttribute(span, enduserRoleAttribute, user.role)
+	}
+
+	if len(user.scope) > 0 {
+		addSpanAttribute(span, enduserScopeAttribute, user.scope)
+	}
+
+	if len(user.session) > 0 {
+		addSpanAttribute(span, enduserSessionAttribute, user.session)
+	}
+}
+
+func addSpanAttribute(span pdata.Span, key string, value string) {
+	attributes := span.Attributes()
+	// don't overwrite existing attributes
+	if _, ok := attributes.Get(key); ok {
+		return
+	}
+	attributes.Insert(key, pdata.NewAttributeValueString(value))
+}
+
+// only capture the user info if all conditions
+// are true.  If an conditions key does not exist
+// in the span, that is considered a failed condition
+func (p *processor) passesConditions(span pdata.Span, conditions []Condition) bool {
+	attribMap := span.Attributes()
+
+	for _, condition := range conditions {
+		attrib, ok := attribMap.Get(condition.Key)
+		if !ok {
+			return false
+		}
+
+		value := attrib.StringVal()
+		// TODO is it correct? it was `value == nil` before
+		if value == "" {
+			return false
+		}
+
+		matched, err := regexp.MatchString(condition.Regex, value)
+		if err != nil {
+			p.logger.Warn("Could not evaluate enduser condition", zap.Error(err))
+			return false
+		}
+		if !matched {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (p *processor) idCapture(value string) *user {
+	return &user{id: value}
+}
+
+func (p *processor) roleCapture(value string) *user {
+	return &user{role: value}
+}
+
+func (p *processor) scopeCapture(value string) *user {
+	return &user{scope: value}
+}
+
+func (p *processor) authHeaderCapture(enduser config, value string) *user {
+	lcValue := strings.ToLower(value)
+	if strings.HasPrefix(lcValue, "bearer ") {
+		tokenString := value[7:]
+		var user *user
+		switch enduser.Encoding {
+		case "jwt":
+			user = p.decodeJwt(enduser, tokenString)
+			// by default use the jwt token as the session id
+			p.setSession(enduser, user, tokenString)
+		default:
+			p.logger.Info("Unknown auth header encoding", zap.String("encoding", enduser.Encoding))
+			return nil
+		}
+		return user
+	} else if strings.HasPrefix(lcValue, "basic ") {
+		token := value[6:]
+		str, err := base64.StdEncoding.DecodeString(token)
+		if err != nil {
+			p.logger.Info("Could not decode basic token", zap.String("value", value))
+			return nil
+		}
+		creds := bytes.SplitN(str, []byte(":"), 2)
+		if len(creds) != 2 {
+			p.logger.Info("Invalid basic token", zap.String("value", value))
+			return nil
+		}
+
+		user := new(user)
+		user.id = string(creds[0])
+		return user
+	} else {
+		p.logger.Info("Authorization header must be basic or bearer", zap.String("value", value))
+	}
+
+	return nil
+}
+
+func (p *processor) jsonCapture(enduser config, value string) *user {
+	v := p.getJSON(value)
```

Comment:
this is where it would be great to re-use the previously parsed json

Reply:
But this field might not be reducted right?